algorithms/search/interpolation_search.py

```diff
@@ -0,0 +1,18 @@
+
+def interpolation_search(arr, x):
+    lo = 0
+    hi = len(arr) - 1
+
+    while lo <= hi and x >= arr[lo] and x <= arr[hi]:
+        m = lo + int(float(hi - lo) / (arr[hi] - arr[lo] + 1) * (x - arr[lo]))
```

Comment:
```suggestion
        m = lo + int((hi - lo) / (arr[hi] - arr[lo] + 1) * (x - arr[lo]))
```
I think removing ` float` improves performance and prevents the additional cascading.